File: src/main/java/net/lomeli/trophyslots/client/config/TrophySlotsFactory.java
package net.lomeli.trophyslots.client.config;

import java.util.Set;

import net.minecraft.client.Minecraft;
import net.minecraft.client.gui.GuiScreen;

import cpw.mods.fml.client.IModGuiFactory;

public class TrophySlotsFactory implements IModGuiFactory {
    @Override
    public void initialize(Minecraft minecraftInstance) {

    }

    @Override
    public Class<? extends GuiScreen> mainConfigGuiClass() {
        return GuiTrophySlotsConfig.class;
    }

    @Override
    public Set<RuntimeOptionCategoryElement> runtimeGuiCategories() {
        return null;
    }

    @Override
    public RuntimeOptionGuiHandler getHandlerFor(RuntimeOptionCategoryElement element) {
        return null;
    }
}

File: src/main/kotlin/net/lomeli/trophyslots/client/config/TrophySlotsFactory.kt
package net.lomeli.trophyslots.client.config

import net.minecraft.client.Minecraft
import net.minecraft.client.gui.GuiScreen
import net.minecraftforge.fml.client.IModGuiFactory

class TrophySlotsFactory : IModGuiFactory {
    override fun runtimeGuiCategories(): MutableSet<IModGuiFactory.RuntimeOptionCategoryElement>? = null

    override fun initialize(minecraftInstance: Minecraft?) {
    }

    override fun hasConfigGui(): Boolean = true

    override fun createConfigGui(parentScreen: GuiScreen?): GuiScreen = GuiTrophySlotsConfig(parentScreen!!)
}